ttnn/cpp/pybind11/pytensor.cpp

```diff
@@ -472,47 +476,64 @@ Tensor convert_python_tensors_to_tt_tensors(
 
 template <typename T>
 owned_buffer::Buffer<T> create_row_major_owned_buffer(
-    owned_buffer::Buffer<T> owned_buffer, const ttnn::TensorSpec& tensor_spec) {
-    if (tensor_spec.layout() == Layout::TILE) {
-        auto data = tensor_impl::convert_layout_tile_to_row_major(
-            tensor_spec.physical_shape(), tensor_spec.tile(), owned_buffer);
-        return owned_buffer::create(std::move(data));
+    owned_buffer::Buffer<T> owned_buffer, const ttnn::TensorSpec& tensor_spec, const bool legacy_output) {
+    if (tensor_spec.memory_config().is_sharded()) {
+        TT_FATAL(
+            tensor_spec.memory_config().shard_spec.has_value(),
+            "Sharded tensors must have a shard spec when converting from tt tensors!");
+    }
+
+    if (legacy_output) {
+        if (tensor_spec.layout() == Layout::TILE) {
+            auto data = tensor_impl::convert_layout_tile_to_row_major(
+                tensor_spec.physical_shape(), tensor_spec.tile(), owned_buffer);
+            return owned_buffer::create(std::move(data));
+        }
+        return owned_buffer;
     }
-    return owned_buffer;
+
+    auto physical_data = owned_buffer.get();
+
+    // Convert physical data based on tensor spec to row major logical data
```

Comment:
Ditto this comment does not provide value

Reply:
Why not? Physical data is described by tensor spec then convert to row major logical data that we can use on host

Reply:
I mean, the code spells the same thing:
```
auto logical_data = tensor_impl::convert_physical_data_to_logical_data(physical_data, tensor_spec);
```